### ISSUE TITLE
Throw error when using postcss-reporter within webpack

### ISSUE THREAD

**Issue by martinkadlec0** [stylelint/stylelint#2572]
I added stylelint to my webpack2 postcss config. Everything works fine except that when any rule is broken I get not only the nice output from postcss-reporter, but also lots of warnings from webpack:
```
WARNING in ./~/css-loader?{"modules":true,"localIdentName":"[name]--[local]--[hash:base64:8]"}!./~/postcss-loader/lib?{"plugins":[null,null,null]}!./client/app/app.css
(Emitted value instead of an instance of Error) stylelint: D:\Work\stack\client\app\app.css:2:2: Expected indentation of 2 spaces (indentation)
 @ ./client/app/app.css 4:14-142 18:2-22:4 19:20-148
 @ ./client/app/app-component.jsx
 @ ./client/app/app-container.jsx
 @ ./client/main.jsx
 @ multi react-hot-loader/patch webpack-hot-middleware/client babel-polyfill ./client/main.jsx
```

> Which rule, if any, is this issue related to?

All of them

> What CSS is needed to reproduce this issue?

Any that breaks any of the rules.

> What stylelint configuration is needed to reproduce this issue?

I use only the stylelint-config-standard config.

```json
module.exports = {
	"extends": "stylelint-config-standard",
	"rules": {
	},
};
```

> Which version of stylelint are you using?

7.10.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

I use webpack2 and webpack-dev-middleware

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope

> What did you expect to happen?

There should be no webpack warnings.

> What actually happened (e.g. what warnings or errors you are getting)?

There is bunch of them.

**Comment by hudochenkov** [comment]
Could you try https://github.com/JaKXz/stylelint-webpack-plugin instead of postcss-loader+stylelint combo?

**Comment by martinkadlec0** [comment]
Tried it and it fixes the problem. Personally though, I prefer the postcss-loader+stylelint combo as it seems more explicit and more likely to work together with other plugins. Is using the plugin the prefered way to use stylelint with webpack?

**Comment by jeddy3** [comment]
> Is using the stylelint-webpack-plugin the prefered way to use stylelint with webpack?

Yes, as it provides access to stylelint's built-in [formatters](https://github.com/stylelint/stylelint/tree/master/lib/formatters). These have better handling of warnings specific to stylelint, like deprecations and invalid option warnings. 

Having said that, you should still be able to use the `postcss-loader`. Have you looked into the `clearAllMessages (boolean, default = false)`, `clearReportedMessages (boolean, default = false)` and `throwError (boolean, default = false)` [options](https://github.com/postcss/postcss-reporter#options) of `postcss-reporter`?

**Comment by martinkadlec0** [comment]
Actually the warnings show regardless of using postcss-reporter, so even if some config of the reporter would fix this I think this should be fixed in the loader plugin itself O:)
Anyhow, since the webpack plugin is the better way to go I switched to that. Thanks for your help!

**Comment by DanielRuf** [comment]
I have the same issue but I use postcss-loader with webpack 3 and want to keep postcss(-loader), is there a solution for this?

**Comment by RehanSaeed** [comment]
I'm also interested in using postcss-loader and stylelint with webpack just to cut out another dependency in my `package.json` file. Has anyone had this combination work for them?

**Comment by DanielRuf** [comment]
@RehanSaeed did you try https://github.com/stylelint/stylelint/blob/master/docs/user-guide/postcss-plugin.md#example-b? We moved linting to separate scripts and decoupled it from the webpack build process in several projects.

**Comment by RehanSaeed** [comment]
@DanielRuf When I try adding stylelint to the postcss pipeline, Webpack just hangs on me. Here is my full webpack file:

```javascript
const path = require('path');
const webpack = require('webpack');
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const CheckerPlugin = require('awesome-typescript-loader').CheckerPlugin;
const stylelint = require("stylelint");

const cssDirectoryName = 'css';
const jsDirectoryName = 'js';

const wwwRootDirectory ='./wwwroot';
const jsDirectory = `./wwwroot/${jsDirectoryName}`;

module.exports = (env) => {
  const isDevBuild = !(env && env.prod);

  function getSassLoaders() {
    let loaders = isDevBuild ? ['style-loader'] : [];
    loaders = loaders
      .concat([
        'css-loader',
        {
          loader: 'postcss-loader', // Run post-css actions
          options: {
            plugins: function () { // post-css plugins, can be exported to postcss.config.js
              return [
                require('precss'),
                require('stylelint')({
                  configFile: '.stylelintrc.json'
                }),
                require('autoprefixer')
              ]
              .concat(isDevBuild ? [] : require('cssnano'));
            }
          }
        },
        'sass-loader'
      ]);
    return loaders;
  }

  return [{
    stats: { modules: false },
    context: __dirname,
    resolve: {
      alias: {
        'vue': 'vue/dist/vue.common.js'
      },
      extensions: ['.js', '.ts']
    },
    entry: { 'main': './ClientApp/Boot.ts' },
    module: {
      rules: [
        {
          test: /\.vue\.html$/,
          include: /ClientApp/,
          loader: 'vue-loader',
          options: {
            loaders: {
              js: 'awesome-typescript-loader?silent=true'
            }
          }
        },
        {
          test: /\.ts$/,
          include: /ClientApp/,
          use: 'awesome-typescript-loader?silent=true'
        },
        {
          test: /\.ts$/,
          enforce: 'pre',
          loader: 'tslint-loader',
          options: {
            fix: true
          }
        },
        {
          test: /\.scss$/,
          use: isDevBuild ? getSassLoaders() : ExtractTextPlugin.extract({ use: getSassLoaders() })
        },
        {
          test: /\.(png|jpg|jpeg|gif|svg)$/,
          use: 'url-loader?limit=25000'
        }
      ]
    },
    output: {
      path: path.join(__dirname, wwwRootDirectory),
      filename: 'js/[name].js',
      publicPath: `${jsDirectoryName}/` // Needs a trailing slash for hot module replacement to work
    },
    plugins: [
      new CheckerPlugin(),
      new webpack.DefinePlugin({
        'process.env': {
          NODE_ENV: JSON.stringify(isDevBuild ? 'development' : 'production')
        }
      }),
      new webpack.DllReferencePlugin({
        context: __dirname,
        manifest: require(`${jsDirectory}/vendor-manifest.json`)
      })
    ].concat(isDevBuild ? [
      // Plugins that apply in development builds only
      new webpack.SourceMapDevToolPlugin({
        filename: '[file].map', // Remove this line if you prefer inline source maps
        moduleFilenameTemplate: path.relative(jsDirectory, '[resourcePath]') // Point sourcemap entries to the original file locations on disk
      })
    ] : [
        // Plugins that apply in production builds only
        new webpack.optimize.UglifyJsPlugin(),
        new ExtractTextPlugin(`${cssDirectoryName}/[name].css`)
      ])
  }];
};
```
